diff --git a/atk/Makefile.am b/atk/Makefile.am
index 0572324..a476cb2 100644
--- a/atk/Makefile.am
+++ b/atk/Makefile.am
@@ -112,7 +112,7 @@ atk-enum-types.h: s-enum-types-h
 s-enum-types-h: @REBUILD@ $(atk_headers) Makefile
 	( cd $(srcdir) && $(GLIB_MKENUMS) \
 			--fhead "#if defined(ATK_DISABLE_SINGLE_INCLUDES) && !defined (__ATK_H_INSIDE__) && !defined (ATK_COMPILATION)\n#error \"Only <atk/atk.h> can be included directly.\"\n#endif\n\n#ifndef __ATK_ENUM_TYPES_H__\n#define __ATK_ENUM_TYPES_H__\n\n#include <glib-object.h>\n\nG_BEGIN_DECLS\n" \
-			--fprod "/* enumerations from \"@filename@\" */\n" \
+			--fprod " /* enumerations from \"@filename@\" */\n" \
 			--vhead "GType @enum_name@_get_type (void);\n#define ATK_TYPE_@ENUMSHORT@ (@enum_name@_get_type())\n"         \
 			--ftail "G_END_DECLS\n\n#endif /* __ATK_ENUM_TYPES_H__ */" \
 		$(atk_headers) ) > tmp-atk-enum-types.h \
