diff --git a/configure.in b/configure.in
index 9d64dfd..6e63690 100644
--- a/configure.in
+++ b/configure.in
@@ -871,43 +871,49 @@ if test "$with_threads" = "no" ; then
 else
     echo Enabling multithreaded support
 
-    AC_CHECK_HEADER(pthread.h,
-	AC_CHECK_LIB(pthread, pthread_join,[
-	   THREAD_LIBS="-lpthread"
-	   AC_DEFINE([HAVE_LIBPTHREAD], [], [Define if pthread library is there (-lpthread)])
-	   AC_DEFINE([HAVE_PTHREAD_H], [], [Define if <pthread.h> is there])
-	   WITH_THREADS="1"]))
     case $host_os in
        *mingw32*) WITH_THREADS="1"
            THREADS_W32="Win32"
-	   THREAD_CFLAGS="$THREAD_CFLAGS -DHAVE_WIN32_THREADS"
+           THREAD_CFLAGS="$THREAD_CFLAGS -DHAVE_WIN32_THREADS"
        ;;
-       *cygwin*) THREAD_LIBS=""
-       ;;
-       *beos*) WITH_THREADS="1"
-	   THREAD_CFLAGS="$THREAD_CFLAGS -DHAVE_BEOS_THREADS"
-       ;;
-       *linux*)
-           if test "${GCC}" = "yes" ; then
-	       GCC_VERSION=`${CC} --version | head -1 | awk '{print $3}'`
-	       GCC_MAJOR=`echo ${GCC_VERSION} | sed 's+\..*++'`
-	       GCC_MEDIUM=`echo ${GCC_VERSION} | sed 's+[[0-9]]*\.++' | sed 's+\..*++'`
-	       if test "${THREAD_LIBS}" = "-lpthread" ; then
-	           if expr ${GCC_MEDIUM} \> 2 \& ${GCC_MAJOR} = 3 > /dev/null
-		   then
-		       THREAD_LIBS=""
-		       BASE_THREAD_LIBS="-lpthread"
-		   else
-		   if expr ${GCC_MAJOR} \> 3 > /dev/null 
-		   then
-		       THREAD_LIBS=""
-		       BASE_THREAD_LIBS="-lpthread"
-		   else
-		       echo old GCC disabling weak symbols for pthread
-		   fi
-		   fi
-	       fi
-	   fi
+       *)   dnl pthread conflicts with win32 threads, so do not look for it there.
+
+            AC_CHECK_HEADER(pthread.h,
+                AC_CHECK_LIB(pthread, pthread_join,[
+                   THREAD_LIBS="-lpthread"
+                   AC_DEFINE([HAVE_LIBPTHREAD], [], [Define if pthread library is there (-lpthread)])
+                   AC_DEFINE([HAVE_PTHREAD_H], [], [Define if <pthread.h> is there])
+                   WITH_THREADS="1"]))
+
+            case $host_os in
+               *cygwin*) THREAD_LIBS=""
+               ;;
+               *beos*) WITH_THREADS="1"
+                   THREAD_CFLAGS="$THREAD_CFLAGS -DHAVE_BEOS_THREADS"
+               ;;
+               *linux*)
+                   if test "${GCC}" = "yes" ; then
+                       GCC_VERSION=`${CC} --version | head -1 | awk '{print $3}'`
+                       GCC_MAJOR=`echo ${GCC_VERSION} | sed 's+\..*++'`
+                       GCC_MEDIUM=`echo ${GCC_VERSION} | sed 's+[[0-9]]*\.++' | sed 's+\..*++'`
+                       if test "${THREAD_LIBS}" = "-lpthread" ; then
+                           if expr ${GCC_MEDIUM} \> 2 \& ${GCC_MAJOR} = 3 > /dev/null
+                           then
+                               THREAD_LIBS=""
+                               BASE_THREAD_LIBS="-lpthread"
+                           else
+                           if expr ${GCC_MAJOR} \> 3 > /dev/null
+                           then
+                               THREAD_LIBS=""
+                               BASE_THREAD_LIBS="-lpthread"
+                           else
+                               echo old GCC disabling weak symbols for pthread
+                           fi
+                           fi
+                       fi
+                   fi
+               ;;
+            esac
        ;;
     esac
     if test "$WITH_THREADS" = "1" ; then
