diff -Nau3r -p zlib-1.2.3-orig/win32/Makefile.gcc zlib-1.2.3/win32/Makefile.gcc
--- zlib-1.2.3-orig/win32/Makefile.gcc	Sat Aug  2 04:49:00 2003
+++ zlib-1.2.3/win32/Makefile.gcc	Sun Aug  2 18:07:42 2009
@@ -25,7 +25,7 @@
 
 STATICLIB = libz.a
 SHAREDLIB = zlib1.dll
-IMPLIB    = libzdll.a
+IMPLIB    = libz.dll.a
 
 #LOC = -DASMV
 #LOC = -DDEBUG -g
@@ -110,6 +110,7 @@ install: zlib.h zconf.h $(LIB)
 	-$(INSTALL) zconf.h $(INCLUDE_PATH)
 	-$(INSTALL) $(STATICLIB) $(LIBRARY_PATH)
 	-$(INSTALL) $(IMPLIB) $(LIBRARY_PATH)
+	-$(INSTALL) $(SHAREDLIB) $(BIN_PATH)
 
 uninstall:
 	-$(RM) $(INCLUDE_PATH)/zlib.h
